contrib/soap/example.js

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright (C) 2024-present  VMaNGOS  https://github.com/vmangos
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+/*
+ * Using https://www.npmjs.com/package/soap does not work because it expects
+ * a WSDL file, which we don't have. Instead, we can use a regular HTTP client
+ * (like the native `fetch` API which is stable since Node.js v21) to construct
+ * and send SOAP requests manually.
+ */
+
+const username = 'ADMINISTRATOR'
+const password = 'ADMINISTRATOR'
+const host = 'localhost'
+const port = 7878
+const command = 'server info'
+
+const buildSoapRequest = command => `
+  <SOAP-ENV:Envelope xmlns:SOAP-ENV="http://schemas.xmlsoap.org/soap/envelope/" xmlns:ns1="urn:MaNGOS">
+    <SOAP-ENV:Body>
+      <ns1:executeCommand>
+        <command>${command}</command>
+      </ns1:executeCommand>
+    </SOAP-ENV:Body>
+  </SOAP-ENV:Envelope>
+`
+
+const soapRequest = buildSoapRequest(command)
+
+;(async () => {
+  try {
+    const response = await fetch(`http://${host}:${port}`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'text/xml',
+        'SOAPAction': 'urn:MaNGOS',
+        'Authorization': 'Basic ' + Buffer.from(`${username}:${password}`).toString('base64')
+      },
+      body: soapRequest
+    })
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`)
+    }
+
+    const result = await response.text()
```

Comment:
Maybe isolate the result, just the php library does.
```suggestion
    const rawResult = await response.text()
    const resultBeginIdx = rawResult.indexOf('<result>') + '<result>'.length
    const resultEndIdx = rawResult.lastIndexOf('</result>')
    const result = rawResult.slice(resultBeginIdx, resultEndIdx)
    console.log('Command succeeded! Output:', result)
```
---
Although I said we dont need any library.
Maybe we can get rid all the HTML encoded entities (like `&#xD;`) with an npm library like [this](https://www.npmjs.com/package/entities).
Or we leave it up the user of this project to figure this out.

Reply:
And maybe we should also ensure that the status code is `200` before doing some string manipulation.

Reply:
I've made adjustments, let me know what you think.